### PR TITLE
feat: article fluxcd image update (FR and EN)

### PR DESCRIPTION
**Pull request by rguichard**

### CODE REVIEW

File: content/blog/flux-auto-image-update.en.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+- [Kubernetes Kustomize with FluxCD](https://particule.io/en/blog/flux-kustomize/)
+- [Canary Deployment with ArgoCD](https://particule.io/en/blog/argocd-canary/)
+- [Flux joins the
+  CNCF](https://particule.io/en/blog/weave-flux-cncf-incubation/)
+
+## Quick reminder about Gitops
+
+Gitops is a set of tools and practices to deploy your applications and your
+infrastructure. Anything defined as code really. This code is stored in a Git
+repository and act as your **single source of truth**. Gitops mechanisms will
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: act -> acts

File: content/blog/flux-auto-image-update.en.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+ensure that the real state (the one in your Kubernetes cluster for example)
+matched your **desired state** (the one in Git) and will reconcile both at all
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: matched -> matches ?
the one [stored | defined] in Git

File: content/blog/flux-auto-image-update.en.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+Why is it good ?
+
+First, you don't need to operate your cluster with `kubectl`, the Gitops agent
+(we'll talk about implementations later) will automatically apply your changes
+as soon as they are committed to git. Git will be your only tool to interact
+with Kubernetes. Code review, PR, MR etc, will allow you to precisely control
+what is applied to your cluster.
+
+This are the 4 principles of Gitops:
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: This -> These ?

File: content/blog/flux-auto-image-update.en.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+- Your system is defined declaratively
+- The desired state of yout system is stored in Git which is your single source
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: yout -> your

File: content/blog/flux-auto-image-update.en.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+  desired state
+
+The two leading Gitops solutions are [FluxCD](https://fluxcd.io/) and
+[ArgoCD](https://argoproj.github.io/argo-cd/). Both are members of the CNCF.
+FluxCD follows the specifications of the Gitops toolkit while ArgoCD follows
+those of the Gitops Engine.
+
+As we already talked about Gitops (in this article and the previous ones), we
+won't come back on those principles and we are going to present a new feature
+in FluxCD v2 : **the imgage update feature**.
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: imgage -> image

File: content/blog/flux-auto-image-update.en.md [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+
+This feature allows Flux to track the changes of your container image into a
+container registry and apply those changes into your Kubernetes cluster.
+
+## Didn't it already exist ?
+
+Yes indeed. This feature exists in FluxCD v1 but was removed in FluxCD v2.
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md:61]: exists -> existed

File: content/blog/flux-auto-image-update.en.md [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+
+Tout à fait, mais dans FluxCD v1, depuis la v2 cette fonction avait disparu
+mais faisait parti des priorités sur la roadmap.
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md:64]: remove ?

File: content/blog/flux-auto-image-update.en.md [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+
+This feature, like the one in FluxCD v1, will track new versions of your image
+and will update your desired state in Git with the new image. You can decide to
+track every new images or only the ones matching a semver rule or a regex. Once
+your desired state is updated, FluxCD reconciliation process will trigger an
+update on your Kubernetes cluster.
+
+One could consider that this feature is out of the scope of Gitops since your
+container registry can be considered as a second source of truth. This is why
+you won't find this feature with ArgoCD.
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md:74]: in ArgoCD ?

File: content/blog/flux-auto-image-update.en.md [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+
+This feature is now back in FlucCD v2 in alpha.
+
+## Let's play !
+
+Let's bootstrap a Flux environment then we will update it to enable the image
+update feature.
+
+You will need a GitHub personnal token with all `repo` access permissions:
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md:83]: personal

File: content/blog/flux-auto-image-update.en.md [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+► configuring deploy key
+✔ deploy key configured
+► generating sync manifests
+► applying sync manifests
+◎ waiting for cluster sync
+✔ bootstrap finished
+```
+
+Unlike Flux v1 which was composed of one controller, Flux v2 uses the Gitops
+toolkit which contains many controllers and can be configure using CRD:
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md:141]: configure -> configured

File: content/blog/flux-auto-image-update.en.md [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+```
+
+I choose to track every versions between `1.0.0` and `2.0.0`. Only
+`particule/helloworld:1.0.0` and `particule/helloworld:1.1.0` should be concerned
+by the update, `particule/helloworld:2.0.0` shouldn't.
+
+Those CRD can set the tracking up but they will not trigger an update into the
+Git repository.
+
+You can check that the tracking work as expected:
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: work -> works ?

File: content/blog/flux-auto-image-update.en.md [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+    authorEmail: fluxcdbot@particule.io
+    authorName: fluxcdbot
+    messageTemplate: 'update image'
+  interval: 1m0s
+  update:
+    strategy: Setters
+```
+
+Check that the Deploy Key in your Git repository has write permission. If that
+is not the cas, you can get the public key used and reconfigure your Github
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.en.md]: cas -> case

File: content/blog/flux-auto-image-update.fr.md [hunk 13]
```diff
@@ -0,0 +1,330 @@
...
+  FluxCD](https://particule.io/blog/flux-semver/)
+- [Continuous delivery avec FluxCD et
+  Concourse-CI](https://particule.io/blog/cicd-concourse-flux/)
+
+## Petit rappel sur Gitops
+
+Gitops est un ensemble d'outil et de pratiques pour déployer vos applications
+et infrastructures. Grâce à l'infrastructure as code, votre infrastructure
+comme vos applications sont définis par du code stocké dans Git. Gitops utilise
+ses fiches déclaratifs comme votre **source unique de vérité**. Il permet
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.fr.md]: fiches -> fichiers ?

File: content/blog/flux-auto-image-update.fr.md [hunk 14]
```diff
@@ -0,0 +1,330 @@
...
+d'assurer une synchronisation parfaite entre l'état désiré (stocké dans Git) et
+l'état réel (typiquement votre cluster Kubernetes).
+
+Pourquoi c'est bien ?
+
+Tout d'abord vous n'avez plus à vous soucier de gérer votre cluster avec
+`kubectl` l'agent Gitops (on reviendra sur les solutions ensuite) se charge
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.fr.md]: `kubectl` l'agent -> `kubectl`: l'agent

File: content/blog/flux-auto-image-update.fr.md [hunk 15]
```diff
@@ -0,0 +1,330 @@
...
+
+```console
+$ docker push particule/helloworld:2.0.0
+```
+
+## Conclusion
+
+Alors oui, c'est exactement comme avant. Sauf qu'on est heureux d'annoncer que
+la fonctionnalité est de nouveau disponible avec les nouvelles CRD associées.
+Peut être n'aviez vous pas mis à jour Flux vers Flux v2 justement parce que
```
Comment (tbobm) [inline content/blog/flux-auto-image-update.fr.md]: Peut être -> Peut-être

**Review by tbobm — CHANGES_REQUESTED** [review]

**Review by ArchiFleKs — COMMENTED** [review]
LGTM, beside the changes proposed by @tbobm

**Review by ArchiFleKs — APPROVED** [review]

**Pull request merged**